src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs

```diff
@@ -106,9 +106,20 @@ public static bool IsPrivilegedProcess
         public static bool IsReleaseLibrary(Assembly assembly) => !IsDebuggable(assembly);
         public static bool IsDebugLibrary(Assembly assembly) => IsDebuggable(assembly);
 
-        // For use as needed on tests that time out when run on a Debug runtime.
+        // For use as needed on tests that time out when run on a Debug or Checked runtime.
         // Not relevant for timeouts on external activities, such as network timeouts.
-        public static int SlowRuntimeTimeoutModifier = (PlatformDetection.IsDebugRuntime ? 5 : 1);
+        public static int SlowRuntimeTimeoutModifier
+        {
+            get
+            {
+                if (IsReleaseRuntime)
+                    return 1;
+                if (IsRiscV64Process)
+                    return IsDebugRuntime? 10 : 2;
+                else
+                    return IsDebugRuntime? 5 : 1;
```

Comment:
We should most likely include "checked" check here too
```suggestion
                    return IsDebugRuntime || IsCheckedRuntime ? 10 : 2;
                else
                    return IsDebugRuntime ? 5 : 1;
```